generators/cloud-init.go

```diff
@@ -107,28 +107,45 @@ func (g *cloudInit) RunLXD(img *image.LXDImage, target shared.DefinitionTargetLX
 
 	switch g.defFile.Name {
 	case "user-data":
-		content = `{{ config_get("user.user-data", properties.default) }}
-`
+		content = `{%- if config_get("cloud-init.user-data", properties.default) == properties.default -%}
+{{ config_get("user.user-data", properties.default) }}
+{%- else -%}
+{{- config_get("cloud-init.user-data", properties.default) }}
+{%- endif -%}`
 		properties["default"] = `#cloud-config
 {}`
 	case "meta-data":
 		content = `instance-id: {{ container.name }}
 local-hostname: {{ container.name }}
+{%- if config_get("cloud-init.meta-data", "") == "" -%}
 {{ config_get("user.meta-data", "") }}
+{%- else -%}
+{{- config_get("cloud-init.meta-data", "") }}
+{%- endif -%}
 `
 	case "vendor-data":
-		content = `{{ config_get("user.vendor-data", properties.default) }}
+		content = `{%- if config_get("cloud-init.user-data", properties.default) == properties.default -%}
+{{ config_get("user.user-data", properties.default) }}
+{%- else -%}
+{{- config_get("cloud-init.user-data", properties.default) }}
+{%- endif -%}
 `
 		properties["default"] = `#cloud-config
 {}`
 	case "network-config":
-		content = `{% if config_get("user.network-config", "") == "" %}version: 1
+		content = `{%- if config_get("cloud-init.network-config", "") == "" -%}
+{{ config_get("user.network-config", "") }}
+{%- else -%}
+{{- config_get("cloud-init.network-config", "") }}
+{%- endif -%}
+`
+		properties["default"] = `version: 1
 config:
   - type: physical
     name: {% if instance.type == "virtual-machine" %}enp5s0{% else %}eth0{% endif %}
     subnets:
       - type: {% if config_get("user.network_mode", "") == "link-local" %}manual{% else %}dhcp{% endif %}
-        control: auto{% else %}{{ config_get("user.network-config", "") }}{% endif %}
+        control: auto
```

Comment:
We can drop the whole user.network-mode thing, that's from LXD 2.0.
Let's just always put dhcp